### DISCUSSION TITLE
Support OpenAI image generation observation thru observe decorator (without as_type)

### DISCUSSION THREAD

**Discussion by fandujar** [langfuse/langfuse#4852] [Ideas]
### Describe the feature or potential improvement

Extend the support of the @observe() decorator to OpenAI image generation without the need of the as_type argument, replicating the same functionality that already exists for chat completions and streams.

Code example:
```
@observe()
def __call_image_generation(model="dall-e-3", prompt="A beautiful image of a cat"):
    client = openai.OpenAI(
        base_url=openai_base_url,
        api_key=openai_api_key,
    )

    return client.images.generate(
        model=model,
        prompt=prompt,
        size="1024x1024",
        quality="standard",
        n=1,
    )
```

The result would be similar to the image below:
![Screenshot 2024-12-30 at 19 09 38](https://github.com/user-attachments/assets/258f663f-ecff-4bae-b34b-23e3373a3128)

As it is today, this is the result:
![Screenshot 2024-12-30 at 19 13 27](https://github.com/user-attachments/assets/f800cb34-bbc7-4be9-bfe6-7fa1a9690552)

**Comment by marcklingen** [comment]
Thanks for sharing.

It would be interesting to extend the existing [openai integration](https://langfuse.com/docs/integrations/openai/python/get-started) which is [interoperable](https://langfuse.com/docs/integrations/openai/python/get-started#use-traces) with the observe decorator to support `client.images.generate`.

Does the current support via the decorator work for you or does it lead to other issues?

**Reply by fandujar** [reply]
HI @marcklingen thanks for replying. 
The current support kinda works. I need to inject some infos to be fully covered, but it works.

**Reply by marcklingen** [reply]
You might want to look into this example for more details on how to log arbitrary LLM calls: https://langfuse.com/docs/sdk/python/decorators#log-any-llm-call

Thanks again for creating this thread, let's track interest for this integration here via upvotes/comments.

**Comment by hcavarsan** [comment]
+1 this would be really useful. im currently using it with `as_type` now but native support would help​​​​​​​​​​​​​​​​
